pkgs/applications/science/math/cplex/default.nix

```diff
@@ -0,0 +1,75 @@
+{ stdenv, coreutils, requireFile, patchelf, makeWrapper, openjdk, gtk2, xorg, glibcLocales, releasePath ? null }:
+
+# To use this package, you need to download your own cplex installer from IBM
+# and override the releasePath attribute to point to the location of the file.  
+#
+# Note: cplex creates an individual build for each license which screws
+# somewhat with the use of functions like requireFile as the hash will be
+# different for every user.
+
+stdenv.mkDerivation rec {
+  name = "cplex-${version}";
+  version = "128";
+  
+  src =
+    if builtins.isNull releasePath then
+      throw ''
+        This nix expression requires that the cplex installer is already
+        downloaded to your machine. Get it from IBM and override the
+        releasePath attribute to point to the location of the file. 
+      ''
+    else
+      releasePath;
+
+  nativeBuildInputs = [ coreutils makeWrapper ];
+  buildInputs = [ openjdk gtk2 xorg.libXtst glibcLocales ];
+
+  phases = "patchPhase buildPhase installPhase fixupPhase";
+
+  patchPhase = ''
+    sed -e 's|/usr/bin/tr"|tr"         |' $src > $name
+  '';
+
+  buildPhase = ''
+    sh $name -i silent -DLICENSE_ACCEPTED=TRUE -DUSER_INSTALL_DIR=$out
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    for pgm in $out/opl/bin/x86-64_linux/oplrun $out/opl/bin/x86-64_linux/oplrunjava $out/opl/oplide/oplide $out/cplex/bin/x86-64_linux/cplex $out/cpoptimizer/bin/x86-64_linux/cpoptimizer;
+    do
+      ln -s $pgm $out/bin;
+    done
```

Comment:
You can make this a single command and use some newlines:
```bash
ln -s $out/opl/bin/x86-64_linux/oplrun \
  $out/opl/bin/x86-64_linux/oplrunjava \
  ... \
  $out/bin
```

Reply:
Done.